### PR TITLE
Updated to Behat v3

### PR DESCRIPTION
**Pull request by Ethan0507**
This Pull Request contains commits for:
* Running tests using Behat v3
* Updated parallel run logic to run on different combinations

This PR should help resolve Issue #11

### CODE REVIEW

**Comment by Ethan0507** [comment]
Hey @francisf , request you to kindly assign someone to review and merge this Pull request

File: config/parallel.conf.yml [hunk 1]
```diff
@@ -1,25 +1,31 @@
...
-                    -
-                        browser: "firefox"
-                    -
-                        browser: "safari"
-                    -
-                        browser: "internet explorer"
+                        environments:
+                            -
+                                browser: "Chrome"
+                                browser_version: "85"
```
Comment (francisf) [inline config/parallel.conf.yml]:
```suggestion
                                browser_version: "latest"
```

File: features/bootstrap/FeatureContext.php [hunk 2]
```diff
@@ -37,3 +32,6 @@ public function iShouldSee($string) {
     }
   }
 }
+
+
```
Comment (francisf) [inline features/bootstrap/FeatureContext.php:36]:
```suggestion
```

File: features/bootstrap/FeatureContext.php [hunk 3]
```diff
@@ -37,3 +32,6 @@ public function iShouldSee($string) {
...
+
```
Comment (francisf) [inline features/bootstrap/FeatureContext.php]: whitespace removal.

File: features/bootstrap/FeatureContext.php [hunk 4]
```diff
@@ -17,7 +12,7 @@ public function iAmOnSite($url) {
   public function iSearchFor($searchText) {
     $element = self::$driver->findElement(WebDriverBy::name("q"));
     $element->sendKeys($searchText);
-    $element->submit();
+    $element->sendKeys(WebDriverKeys::ENTER);
```
Comment (francisf) [inline features/bootstrap/FeatureContext.php:15]: why is this required ?

File: lib/BrowserStackContext.php [hunk 5]
```diff
@@ -2,39 +2,40 @@
 
 require 'vendor/autoload.php';
 
-class BrowserStackContext extends Behat\Behat\Context\BehatContext
+class BrowserStackContext implements Behat\Behat\Context\Context
 {
-    protected $CONFIG;
+    protected static $CONFIG;
```
Comment (francisf) [inline lib/BrowserStackContext.php:7]: do we need static?

File: lib/BrowserStackContext.php [hunk 6]
```diff
@@ -2,39 +2,40 @@
...
     protected static $driver;
-    private static $bs_local;
+    protected static $bs_local;
```
Comment (francisf) [inline lib/BrowserStackContext.php]: any reason for protected?

File: lib/BrowserStackContext.php [hunk 7]
```diff
@@ -2,39 +2,40 @@
...
+    public static function createDriver()
     {
-        $CONFIG = $GLOBALS['CONFIG'];
         $task_id = getenv('TASK_ID') ? getenv('TASK_ID') : 0;
-
-        $url = "http://" . $GLOBALS['BROWSERSTACK_USERNAME'] . ":" . $GLOBALS['BROWSERSTACK_ACCESS_KEY'] . "@" . $CONFIG['server'] ."/wd/hub";
-        $caps = $CONFIG['environments'][$task_id];
-
-        foreach ($CONFIG["capabilities"] as $key => $value) {
+        echo "TASK ".$task_id."\n\n";
```
Comment (francisf) [inline lib/BrowserStackContext.php]:
```suggestion
```

File: lib/parallel.php [hunk 8]
```diff
@@ -7,17 +7,17 @@
...
 
 foreach ($CONFIG['environments'] as $key => $value) {
-    if(preg_match('/win/i', PHP_OS)) {
+    if (strtoupper(substr(PHP_OS, 0, 3)) === 'WIN') {
       // Windows
       $cmd = "set TASK_ID=$key& \"./bin/behat\" --config=". getenv("CONFIG_FILE")." 2>&1\n";
     } else {
       // Linux or  Mac
-      $cmd = "TASK_ID=$key ./bin/behat --config=". getenv("CONFIG_FILE")." 2>&1\n";
+      $cmd = "export TASK_ID=$key && ./bin/behat --config=". getenv("CONFIG_FILE")." 2>&1\n";
```
Comment (francisf) [inline lib/parallel.php]: not needed.

File: features/bootstrap/FeatureContext.php (hunk 4 above)
Comment (Ethan0507) [reply features/bootstrap/FeatureContext.php:15]: element.submit() on latest Chrome browsers results to the following error: "unknown command: Cannot call non W3C standard command while in W3C mode (UnknownCommandException)"

File: lib/BrowserStackContext.php (hunk 5 above)
Comment (Ethan0507) [reply lib/BrowserStackContext.php:7]: Yes, since we cannot write to $GLOBALS in the '__construct()' method as of PHP 8.1.0 onwards, we need a static variable.

**Comment by Ethan0507** [comment]
@francisf Performed suggested changes to this PR

**Review by francisf — APPROVED** [review]
